examples/specs/interactive_bar_select_highlight.vl.json

```diff
@@ -0,0 +1,49 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
+  "description": "A bar chart with highlighting on hover and selecting on click. (Inspired by Tableau's interaction style.)",
+  "data": {
+    "values": [
+      {"a": "A","b": 28}, {"a": "B","b": 55}, {"a": "C","b": 43},
+      {"a": "D","b": 91}, {"a": "E","b": 81}, {"a": "F","b": 53},
+      {"a": "G","b": 19}, {"a": "H","b": 87}, {"a": "I","b": 52}
+    ]
+  },
+  "selection": {
+    "highlight": {"type": "single", "empty": "none", "on": "mouseover"},
+    "select": {"type": "multi"}
+  },
+  "mark": {
+    "type": "bar",
+    "fill": "#4C78A8",
+    "stroke": "black",
+    "cursor": "pointer"
+  },
+  "encoding": {
+    "x": {"field": "a", "type": "ordinal"},
+    "y": {"field": "b", "type": "quantitative"},
+    "fillOpacity": {
+      "condition": {"selection": "select", "value": 1},
+      "value": 0.3
+    },
+    "strokeWidth": {
+      "condition": [
+        {
+          "test": {
+            "and": [
+              {"selection": "select"},
+              "length(data(\"select_store\"))"
```

Comment:
Note that this is a cheat by directly accessing the underlying data store to check if the selection is not empty. I think we might wanna expose this as a type of predicate in the future.

Reply:
Don't we have a way to change the selection to be empty or everything when nothing is selected by the user?

Reply:
We do, but it doesn't work for this case since we need the `opacity` to be `1` when the `"select"` selection is empty but the `strokeWidth` to be `0` when the `"select"` selection is empty.

Reply:
Can you make two selections?

Reply:
Nope, that totally doesn't make sense. There should be only one selection for clicking that gets applied to production rule differently.